tools/importer-rest-api-specs/components/parser/models.go

```diff
@@ -526,11 +532,13 @@ func (d SwaggerDefinition) parseObjectDefinition(modelName, propertyName string,
 		// then just return that object instead, there's no point creating the wrapper type
 		if len(input.Properties) == 0 && len(input.AllOf) == 1 {
 			inheritedModel := input.AllOf[0]
-			return d.parseObjectDefinition(inheritedModel.Title, propertyName, &inheritedModel, result)
+			return d.parseObjectDefinition(inheritedModel.Title, propertyName, &inheritedModel, result, true)
 		}
 
 		// check for / avoid circular references
-		if _, ok := result.Models[modelName]; !ok {
+		// only parse model when modelName equals PropertyName, otherwise the `input` Schema may not for the modelName
+		// if parsing a top-level or an inlined model, should pass the propertyName just the same as the modelName
```

Comment:
I think this'd be clearer as:

```suggestion
		// however we should only do this when we're parsing a model (`parsingModel`) directly rather than when parsing a model from a field - and only if we haven't already parsed this model
```